### DISCUSSION TITLE
zod in server?

### DISCUSSION THREAD

**Discussion by woile** [47ng/nuqs#526] [Q&A] [answered]
How can I use `parseAsJson` with `zod` in the server? It doesn't seem to work, I've been trying to do:
`createSearchParamsCache(parseAsJson(MySchema.parse))` but it doesn't seem to work 🤔

**Comment by franky47** [comment]
You need to pass an object to `createSearchParamsCache` where the keys are the names of the search params, and the values are the associated parser.

So if your state was named `foo` (eg: `?foo={"hello":"world"}`), it would look like:

```ts
export const cache = createSearchParamsCache({
  foo: parseAsJson(MySchema.parse)
})
```

**Reply by woile** [reply]
Thanks, but I was trying to parse the queryparams into a zod object. Is this currently possible? like from `foo=1&bar=2` into `{foo: 1, bar: 2}`
Thanks!

**Reply by franky47** [reply, marked as answer]
> Is this currently possible?

Not via `nuqs`, but you could run your schema against the [`searchParams` page prop](https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional):

```ts
export default function Page({ searchParams }) {
  // Error handling omitted for brevity
  const { foo } = MySchema.parse(searchParams)
  // ...
}
```

Using Zod would give you some additional benefits, such as being able to validate combinations of key/values, which `nuqs` doesn't do (it only parses one value at a time with no relationship between keys).
